src/com/facebook/buck/swift/SwiftCompile.java

```diff
@@ -82,29 +83,36 @@
   @AddToRuleKey(stringify = true)
   private final Path outputPath;
 
+  @AddToRuleKey(stringify = true)
   private final Path objectFilePath;
+  @AddToRuleKey(stringify = true)
   private final Path modulePath;
+  @AddToRuleKey(stringify = true)
   private final Path moduleObjectPath;
+  @AddToRuleKey(stringify = true)
   private final ImmutableList<Path> objectPaths;
   private final Optional<AbsPath> swiftFileListPath;
 
   @AddToRuleKey private final boolean shouldEmitSwiftdocs;
+  @AddToRuleKey private final boolean useModulewrap;
+  @AddToRuleKey private final boolean compileForceCache;
+  @AddToRuleKey(stringify = true)
   private final Path swiftdocPath;
 
   @AddToRuleKey private final ImmutableSortedSet<SourcePath> srcs;
   @AddToRuleKey private final SwiftTargetTriple swiftTarget;
   @AddToRuleKey private final Optional<String> version;
   @AddToRuleKey private final ImmutableList<? extends Arg> compilerFlags;
 
+  @AddToRuleKey(stringify = true)
   private final Path headerPath;
-  private final CxxPlatform cxxPlatform;
-  private final ImmutableSet<FrameworkPath> frameworks;
+  @AddToRuleKey private final ImmutableSet<FrameworkPath> frameworks;
+  @AddToRuleKey private final AddsToRuleKeyFunction<FrameworkPath, Path> frameworkPathToSearchPath;
+  @AddToRuleKey private final Flavor flavor;
```

Comment:
@xianwen: This requires `@AddToRuleKey(stringify = true)` as `Flavor` is not a supported type for rulekey computation.

Reply:
Oops, sorry about that, I'll send you another PR fixing it up.

Reply:
https://github.com/facebook/buck/pull/2475

Reply:
Merged, thanks.